subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -79,6 +84,13 @@
 
     private FileCollection annotationProcessorPath;
 
+    private final PropertyState<File> annotationProcessorGeneratedSourcesDirectory;
```

Comment:
Why don't we use a `DirectoryVar` here?

Reply:
We discussed this via voice. FTR, `DirectoryVar` is currently only used in native tasks. Let's start using it when we restructure how compiler options are specified; the current API is chaotic enough as it is.